apps/sms/test/unit/thread_ui_test.js

```diff
@@ -2208,43 +2253,67 @@ suite('thread_ui.js >', function() {
         ThreadUI.renderMessages(1);
       });
 
-      test('infinite rendering test', function() {
+      test('infinite rendering test', function(done) {
         var chunkSize = ThreadUI.CHUNK_SIZE;
         var message;
 
-        for (var i = 1; i < chunkSize; i++) {
-          MessageManager.getMessages.yieldTo(
-            'each', MockMessages.sms({ id: i })
-          );
-          message = document.getElementById('message-' + i);
-          assert.ok(
-            message.classList.contains('hidden'),
-            'message-' + i + ' should be hidden'
-          );
+        function assertHidden() {
+          return new Promise((resolve, reject) => {
+            for (var i = 1; i < chunkSize; i++) {
+              message = document.getElementById('message-' + i);
+              assert.ok(
+                message.classList.contains('hidden'),
+                'message-' + i + ' should be hidden'
+              );
+            }
+
+            MessageManager.getMessages.yieldTo(
+              'each', MockMessages.sms({ id: chunkSize })
+            );
+            TaskRunner.prototype.push.lastCall.returnValue.then(
```

Comment:
nit: maybe if you create suite-level func with descriptive name that returns ```TaskRunner.prototype.push.lastCall.returnValue``` it would be easier to read?